assignment4/assignment4.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <math.h>
+#include <vector>
+
+class IslandCounter{
+	void collapse_island(std::vector< std::vector<bool> > &map, int i, int j){
+		if(!map[i][j]) return;
+
+		map[i][j] = false;
+		collapse_island( map, std::min(i+1, (int) (map.size()-1)), j);
+		collapse_island( map, std::max(i-1, 0), j);
+		collapse_island( map, i, std::min(j+1,(int) (map[i].size()-1)));
+		collapse_island( map, i, std::max(j-1,0));
```

Comment:
Maybe I could try with the iterative implementation of DFS, but I am not sure if that would solve the problem. I would still need a stack for neighbour positions. I guess the algorithm proposed with hint 2 is more efficient but I haven't come up with it yet :/